### ISSUE TITLE
[Bug]: Target Storage empty in the migration dialog when the PVE storage listing exceeds the 8s timeout

### ISSUE THREAD

**Issue by adminsyspro** [adminsyspro/proxcenter-ui#742]
## Summary

Reported during an Enterprise trial: in the "Migrate to Proxmox VE" dialog, the **Target Storage** field stays empty and greyed out, so the Migrate button never enables and the migration cannot be started. Temporary Storage is populated correctly on the same dialog, which makes the field look "locked" rather than broken. The two fields have different sources: Temporary Storage is detected over SSH, Target Storage comes from the Proxmox API.

The trial environment reproduces it on two separate clusters (PVE 8 + Ceph, and another cluster), while a freshly installed Proxmox node works fine. The difference is response time, not configuration: both affected clusters declare a large number of PBS storages.

```
# on the affected node
time pvesh get /nodes/NODE/storage --output-format json
real    0m20.824s
```

Frontend logs at the same moment:

```
[failover] Connection [redacted] failure 1/2 for /nodes/<node>/storage (timeout)
Error fetching storages: Error [TimeoutError]: The operation was aborted due to timeout
```

This is the same failure mode as #332, on a different call site. #332 fixed the migration-time config mutations only, deliberately leaving the global default alone, so read paths were never covered.

## Root cause 1: storage listing runs on the 8s default

`pveFetch` defaults to 8s (`frontend/src/lib/proxmox/client.ts:138`, duplicated at `:169`), hardcoded, with no named constant. The storage route calls it with no `fetchOpts`, so a listing that legitimately takes 20s can never succeed:

```ts
// frontend/src/app/api/v1/connections/[id]/nodes/[node]/storages/route.ts:34
let storages = await pveFetch<any[]>(conn, `/nodes/${encodeURIComponent(node)}/storage`)
```

The timeout is not configurable anywhere: no environment variable, no field on the `Connection` model, no `Setting` key. The only escape hatch is the `fetchOpts.timeoutMs` argument, used at just two call sites in the whole repo (120s for `PUT /qemu/{vmid}/config` from #332, 30s for NFS content listing).

## Root cause 2: the UI swallows the error

`frontend/src/app/(dashboard)/infrastructure/inventory/InventoryDetails.tsx:847-857` ends the storage fetch with an empty catch. It is the only one of the five fetches in that effect whose catch sets no error state:

```tsx
fetch(`/api/v1/connections/${migTargetConn}/nodes/${fetchNode}/storages?content=images`).then(r => r.json()).then(d => {
  const storages = (d.data || d || []).filter((s: any) => {
```

Two defects here, independent of any timeout:

- No `r.ok` check. On a 500 the body is `{ error: "..." }`, so `d.data` is `undefined`, `d` is truthy, and `.filter` throws `TypeError: (d.data || d || []).filter is not a function`. Same on a 403 from the RBAC guard.
- The `TypeError` is swallowed by the empty `.catch(() => {})`, so `setMigStorages` is never reached and `migStorages` keeps its previous value, which can be the stale list from the previously selected node.

Downstream, `InventoryDialogs.tsx:2282` disables the whole `FormControl` when `migStorages` is empty, with no message. The bulk variant (`InventoryDialogs.tsx:3544`) instead leaves the select clickable with an empty menu. The i18n key `inventoryPage.esxiMigration.selectStorage` exists but is never used.

## Root cause 3: a timeout trips the failover circuit breaker

`client.ts` separates `isHardNetworkError` from `isTimeoutError`, and even comments "node may just be slow, not dead", but the failover guard at `client.ts:266` tests the union `isNetworkError`. The distinction has exactly one observable effect: the `(timeout)` vs `(hard error)` wording in the log at `client.ts:292`.

Consequences on a uniformly slow cluster:

- The failure counter is per connection, not per node or per path, and never decays (`frontend/src/lib/cache/nodeIpCache.ts:86-102`). It resets only on success, so it stays pinned at the threshold of 2 while the call keeps failing.
- At the threshold the code scans the other cluster nodes sequentially at 5s each. Every node is equally slow for the same reason, so all candidates fail and the request ends as `all cluster nodes unreachable (tried N nodes)` (`client.ts:376`), which misreports a healthy cluster as down.
- After a failover the breaker stays OPEN for 60s (`client.ts:84`), routing all traffic for that connection to a standby node for a reason unrelated to node health.

## Root cause 4: caller signals can only shorten the timeout

`client.ts:178-180` combines the caller signal and the internal timeout with `AbortSignal.any`, so the effective timeout is the **minimum** of the two. Callers that pass a longer budget silently get 8s:

- `frontend/src/app/api/v1/dashboard/route.ts:195` passes `AbortSignal.timeout(15000)`
- `frontend/src/app/api/v1/resources/overview/route.ts:866` passes `AbortSignal.timeout(15000)`

Both are capped at 8s and do not behave as written. This degrades the dashboard and the resources overview on any slow cluster, independently of the migration dialog.

## Proposed fix

- Introduce a named, centralized timeout for inventory reads that are slow by nature, with an environment variable override, and apply it to the storage listing. Follow the precedent already set for NFS content listing rather than raising the 8s default, which remains appropriate for polling.
- Surface the failure in the migration dialog instead of rendering an empty field: check `r.ok`, stop parsing an error body as a list, and clear stale state on node change.
- Stop letting a timeout trip the failover circuit breaker, or at least exclude the long read paths, since the two error classes are already distinguished in code.
- Fix the `AbortSignal.any` capping so a caller can lengthen, not only shorten, the request budget.

## Affected files

- `frontend/src/lib/proxmox/client.ts`
- `frontend/src/lib/cache/nodeIpCache.ts`
- `frontend/src/app/api/v1/connections/[id]/nodes/[node]/storages/route.ts`
- `frontend/src/app/(dashboard)/infrastructure/inventory/InventoryDetails.tsx`
- `frontend/src/app/(dashboard)/infrastructure/inventory/components/InventoryDialogs.tsx`
